### PR TITLE
Add symbol packages for coreclr runtime.

### PR DESCRIPTION
**Pull request by mikem8361**
Depends on the new buildtools symbol support.

Update build tools version containing the new symbol file support.

### CODE REVIEW

**Comment by mikem8361** [comment]
@gkhanna79, @dagood could you review? Thanks.

File: src/.nuget/descriptions.json [hunk 1]
```diff
@@ -8,5 +8,10 @@
         "Name": "Microsoft.NETCore.Runtime.CoreCLR",
         "Description": "The .NET Core runtime, called CoreCLR, and the base library, called mscorlib. It includes the garbage collector, JIT compiler, base .NET data types and many low-level classes.",
         "CommonTypes": [ ]
+    },
+    {
+        "Name": "Microsoft.NETCore.Runtime.CoreCLR.symbols",
+        "Description": "The .NET Core runtime symbol files",
+        "CommonTypes": [ ]
```
Comment (dagood) [inline src/.nuget/descriptions.json]: Is this used? (I'm imagining it's left over from the previous approach, but maybe I'm wrong.)

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/ubuntu/Microsoft.NETCore.Runtime.CoreCLR.pkgproj [hunk 2]
```diff
@@ -46,5 +47,27 @@
...
 
+  <ItemGroup Condition="'$(__BuildType)' == 'Release'">
+    <ArchitectureSpecificNativeSymbol Include="$(BinDir)libcoreclr.so.dbg"/>
+    <ArchitectureSpecificNativeSymbol Include="$(BinDir)libcoreclrtraceptprovider.so.dbg"/>
+    <ArchitectureSpecificNativeSymbol Include="$(BinDir)libdbgshim.so.dbg"/>
+    <ArchitectureSpecificNativeSymbol Include="$(BinDir)libmscordaccore.so.dbg"/>
+    <ArchitectureSpecificNativeSymbol Include="$(BinDir)libmscordbi.so.dbg"/>
+    <ArchitectureSpecificNativeSymbol Include="$(BinDir)libsos.so.dbg"/>
+    <ArchitectureSpecificNativeSymbol Include="$(BinDir)libsosplugin.so.dbg"/>
+    <ArchitectureSpecificNativeSymbol Include="$(BinDir)System.Globalization.Native.so.dbg"/>
```
Comment (dagood) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/ubuntu/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: Hmm, do you think it would be worth doing something like this to reduce repetition?

```XML
<NativeSplittableBinary Include="$(BinDir)libcoreclr.so" />
<NativeSplittableBinary Include="$(BinDir)libcoreclrtraceptprovider.so" />

<ArchitectureSpecificNativeFile Include="@(NativeSplittableBinary)" />
<ArchitectureSpecificNativeSymbol Include="@(NativeSplittableBinary -> '%(Identity).pdb')"/>
```

**Comment by dagood** [comment]
Looks good on the OSX side, and the result from my build worked on myget: https://www.myget.org/feed/dagood-test-dotnet-core/package/nuget/runtime.osx.10.10-x64.Microsoft.NETCore.Runtime.CoreCLR

As for the CI error, maybe the `__BuildType` check is actually required? In case removing that was a response to my comment, I just meant to show a way of listing the files only once, not removing the condition (although I can see how it could be read that way because I did put those items next to each other).

**Comment by mikem8361** [comment]
@dagood I've addressed all your feedback and the CI builds pass. Should I merge it (after I smash the commits of course)?

**Comment by dagood** [comment]
LGTM! I think it would be good to wait for @gkhanna79 to review too though, since this was my first time working with the coreclr packaging.

**Comment by mikem8361** [comment]
ping @gkhanna79 for review.

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj [hunk 3]
```diff
@@ -38,7 +38,6 @@
     <ProjectReference Include="ubuntu\Microsoft.NETCore.Runtime.CoreCLR.pkgproj">
       <Platform>amd64</Platform>
     </ProjectReference>
-
```
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: Can we undo this change?

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/ubuntu/Microsoft.NETCore.Runtime.CoreCLR.pkgproj [hunk 4]
```diff
@@ -46,5 +45,21 @@
...
+    <AdditionalLibPackageExcludes Include="%2A%2A\%2A.dbg"/>
+    <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.so"/>
+    <AdditionalSymbolPackageExcludes Include="%2A%2A\%2A.dll"/>
+    <AdditionalSymbolPackageExcludes Include="%2A%2A\sosdocsunix.txt"/>
+    <AdditionalSymbolPackageExcludes Include="%2A%2A\crossgen"/>
+    <ArchitectureSpecificNativeSymbol Include="..\_.pdb"/>
+
+    <File Include="@(ArchitectureSpecificNativeSymbol)">
+      <TargetPath>runtimes/$(PackageTargetRuntime)/native</TargetPath>
+      <IsSymbolFile>true</IsSymbolFile>
```
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/ubuntu/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: We are placing the symbols in the same package?

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj (hunk 3 above)
Comment (mikem8361) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: You mean the removal of the blank line? Is the change that big of deal?

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/ubuntu/Microsoft.NETCore.Runtime.CoreCLR.pkgproj (hunk 4 above)
Comment (mikem8361) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/ubuntu/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: No, all of this creates a separate symbol package. Same package id, but different file name (ends in .symbols.nupkg and put in the bin/Product/Linux.x64.Release/.nuget/symbolpkg directory). @dagood can better answer all the reasons why.

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj (hunk 3 above)
Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: It would be helpful to not introduce redundant changes. Your call.

**Comment by gkhanna79** [comment]
Please squash the commits before merging.

File: src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/ubuntu/Microsoft.NETCore.Runtime.CoreCLR.pkgproj (hunk 4 above)
Comment (dagood) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/ubuntu/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: Right, the package gets split up by the path globs in this file plus the ones in buildtools: https://github.com/dotnet/buildtools/blob/master/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetPack.cs#L19-L34

@gkhanna79 Did you want more details why or is that fine?

Comment (gkhanna79) [inline src/.nuget/Microsoft.NETCore.Runtime.CoreCLR/ubuntu/Microsoft.NETCore.Runtime.CoreCLR.pkgproj]: Sounds good to me.
